app/containers/ItunesContainer/tests/saga.test.js

```diff
@@ -49,4 +52,24 @@ describe('ItunesContainer saga tests', () => {
       })
     );
   });
+
+  it('should start task to watch for REQUEST_GET_TRACK_DETAILS action', () => {
+    expect(trackGenerator.next().value).toEqual(
+      takeLatest(itunesContainerTypes.REQUEST_GET_TRACK_DETAILS, getTrackData)
+    );
+  });
+
+  it('should ensure that the action FAILURE_GET_TRACK_DETAILS is dispatched when the api call fails', () => {
+    getTrackDetailsGenerator = getTrackData({ trackId });
+    getTrackDetailsGenerator.next().value;
+    const response = getTrackDetailsGenerator.next().value;
+    expect(response).toEqual(call(getTrackDetails, trackId));
+    const errorRes = translate('something_went_wrong');
+    expect(getTrackDetailsGenerator.next(apiResponseGenerator(false, errorRes)).value).toEqual(
+      put({
+        type: itunesContainerTypes.FAILURE_GET_TRACK_DETAILS,
+        error: errorRes
+      })
+    );
+  });
```

Comment:
Add a test for the untested lines in the saga. Especially the logic related to `find`.